File: JavaScript_Courses/JavaScript_Basics/Preparation/05_While_Loop_Exercise/06_Cake.js
function cake(input) {
    let width = Number(input[0]);
    let length = Number(input[1]);
    let cakePieces = width * length;

    let currentIndex = 2;

    while (true) {
        let currentPiece = input[currentIndex];

        if (currentPiece == "STOP") {
            console.log(`${cakePieces} pieces are left.`);
            break;
        }

        cakePieces -= Number(currentPiece);
        if (cakePieces < 0) {
            console.log(`No more cake left! You need ${-cakePieces} pieces more.`);
            break;
        }

        currentIndex++;
    }
}


// function cake(input) {
//     let width = Number(input[0]);
//     let length = Number(input[1]);
//     let cakePieces = width * length;

//     let currentIndex = 2;
//     let currentPiece = input[currentIndex];
//     while (currentPiece != "STOP") {
//         cakePieces -= Number(currentPiece);
//         if (cakePieces < 0) {
//             console.log(`No more cake left! You need ${-cakePieces} pieces more.`);
//             break;
//         }

//         currentIndex++;
//         currentPiece = input[currentIndex];
//     }

//     if (currentPiece == "STOP") {
//         console.log(`${cakePieces} pieces are left.`);
//     }
// }


// cake([
//     "10",
//     "10",
//     "20",
//     "20",
//     "20",
//     "20",
//     "21"
// ]);

File: Python_Courses/Python_Basics/Python_PB_2024_01/05_While_Loop_Exercise/06_Cake.py
width = int(input())
length = int(input())

cake_pieces = width * length
cake_pieces_left = cake_pieces
is_cake_left = True

command = input()  # either 'STOP', or current_pieces as string, e.g. '20'
while command != 'STOP':
    current_pieces = int(command)
    cake_pieces_left -= current_pieces
    if cake_pieces_left < 0:
        is_cake_left = False
        break

    command = input()

if is_cake_left:
    print(f'{cake_pieces_left} pieces are left.')
else:
    print(f'No more cake left! You need {-cake_pieces_left} pieces more.')
